src/lib/components/AccountInsights.svelte

```diff
@@ -0,0 +1,136 @@
+<script lang="ts">
+  import { convertValue } from '$lib/conversion'
+  import { BitcoinDenomination } from '$lib/types'
+  import { truncateValue } from '$lib/utils'
+  import { onDestroy, onMount } from 'svelte'
+  import { channelsAPY$ } from '$lib/streams'
+  import { translate } from '$lib/i18n/translations'
+  import type { ChannelAPY } from '$lib/backends'
+  import type { ChartItem } from 'chart.js'
+
+  let showZeroFeesMessage = false
+
+  enum ChartIDs {
+    fees = 'fees',
+    apy = 'apy'
+  }
+  const charts = [
+    {
+      id: ChartIDs.fees,
+      title: $translate('app.labels.routing_chart')
+    },
+    {
+      id: ChartIDs.apy,
+      title: $translate('app.labels.apy_chart')
+    }
+  ]
+
+  async function renderFeesChart(channels: ChannelAPY[]) {
+    const { Chart } = await import('chart.js/auto')
+
+    let labels: string[] = []
+    let data: number[] = []
+
+    channels?.forEach(({ account, fees_in_msat, fees_out_msat }) => {
+      const routingFees = fees_in_msat + fees_out_msat
+
+      if (routingFees) {
+        const value = Math.round(
+          Number(
+            convertValue({
+              value: routingFees.toString(),
+              from: BitcoinDenomination.msats,
+              to: BitcoinDenomination.sats
+            })
+          )
+        )
+
+        labels.push(truncateValue(account, 5))
+        data.push(value)
+      }
+    })
+
+    const el: ChartItem | null = document.getElementById(ChartIDs.fees) as ChartItem
+
+    if (el) {
+      new Chart(el, {
+        type: 'pie',
+        data: {
+          labels,
+          datasets: [
+            {
+              data
+            }
+          ]
+        }
+      })
+    }
+  }
+
+  async function renderApyChart(channels: ChannelAPY[]) {
+    const { Chart } = await import('chart.js/auto')
+
+    let labels: string[] = []
+    let data: number[] = []
+
+    channels?.forEach(({ account, apy_total }) => {
+      const total = Number(apy_total.substring(0, apy_total.length - 1))
+
+      if (total) {
+        labels.push(truncateValue(account, 5))
+        data.push(Number(apy_total.substring(0, apy_total.length - 1)))
+      }
+    })
+
+    const el: ChartItem | null = document.getElementById(ChartIDs.apy) as ChartItem
+
+    if (el) {
+      new Chart(el, {
+        type: 'pie',
+        data: {
+          labels,
+          datasets: [
+            {
+              data
+            }
+          ]
+        }
+      })
+    }
+  }
+
+  onMount(async () => {
+    const net = channelsAPY$.value.data?.filter((item) => item.account === 'net')[0]
+
+    if (net && net?.fees_in_msat + net?.fees_out_msat === 0) {
+      showZeroFeesMessage = true
+      return
+    }
+
+    const channels = channelsAPY$.value.data?.filter((item) => item.account !== 'net')
+
+    if (channels) {
+      await renderFeesChart(channels)
+      await renderApyChart(channels)
+    }
+  })
```

Comment:
```suggestion
  onMount(async () => {
    const { net, channels } = (channelsAPY$.value.data || []).reduce(
      (acc, item) => {
        acc[item.account === 'net' ? 'net' : 'channels'].push(item)
        return acc
      },
      { net: [], channels: [] } as { net: ChannelAPY[]; channels: ChannelAPY[] }
    )

    if (net[0] && net[0].fees_in_msat + net[0].fees_out_msat === 0) {
      showZeroFeesMessage = true
      return
    }

    if (channels.length) {
      renderFeesChart(channels)
      renderApyChart(channels)
    }
  })
```

We can use `reduce` here to do a single iteration through the `channelsAPY` data. We can also render the two charts in parallel to speed it up ever so slightly.
With this code though, we have no way of updating the data and the charts, so I think it might be better to put `net` and `channels` into reactive (`$:`) variables that will update when the data updates. Then the rendering functions will need to check if there is a chart already rendered, and if so, then just update the data rather than rendering a whole new chart.